crypto/x509/x_x509.c

```diff
@@ -152,13 +152,18 @@ X509 *X509_new_with_libctx(OPENSSL_CTX *libctx, const char *propq)
     X509 *cert = NULL;
 
     cert = (X509 *)ASN1_item_new((X509_it()));
-    if (cert != NULL) {
-        cert->libctx = libctx;
-        cert->propq = propq;
-    }
+    (void)x509_set0_libctx(cert, libctx, propq);
     return cert;
 }
 
+X509 *x509_dup_with_libctx(const X509 *cert)
```

Comment:
Seems strange to have a "with_libctx" function that doesn't take a libctx as a parameter. Shouldn't this just be "X509_dup" - and that function should do the right thing with the libctx in the src X509 if it is present?

Reply:
Really strange, but at least while extending the CMP msg and client tests with provider support it looked like `X509_dup()` does not copy the `libctx` and `propq` fields of `x509_st` and this appeared to be the reason why the extended tests. I cannot find the actual implementation of that function, likely because it is implemented purely generically as for many other ASN.1-based structures, but I suspect that non-ASN.1-fields are not actually copied.
Still i tried again without the new function  `X509_dup()_with_libctx()`, and for whatever reason it is not actually needed any more. So I removed it again.